### PR TITLE
chore(deployment): add dynamic deployment with now.sh

### PR DESCRIPTION
**Pull request by layershifter**
This PR aims to add automatic deployment our doc site on temporary domains.

Details: https://zeit.co/blog/every-push-now
Sample website: https://stardust-react-p6b8lvin2.now.sh

----

# Plans

Full details: https://zeit.co/account/plan
Bandwidth: 100 GB / mo

We don't use serverless, so I assume that bandwidth is single valid limitation. I also don't see any limitations for build count.

# Workflow

On each commit a new build will be created with Github hooks. If a new commit will be pushed before a build will end, it will be cancelled. I.e. we will have only one build per PR.

Each PR will have an URL with published docsite, it will be available in PR comments.

More details: https://zeit.co/github#features

### CODE REVIEW

File: .nowignore [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+.snyk
+codecov.yml
+dangerfile.ts
+jest.config.js
+lerna-debug.log
+CHANGELOG.md
+LICENSE.md
+MANIFESTO.md
+README.md
+yarn-error.log
```
Comment (layershifter) [inline .nowignore]: This config allow to ignore files for sync.
General process:
- upload files to Now
- build them with Now
- publish

File: build/gulp/plugins/gulp-doctoc.ts [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+import config from '../../../config'
 import sh from '../sh'
 
+const insideGitRepo = fs.existsSync(config.paths.base('.git'))
+
 export default () =>
   through2.obj((file, enc, done) => {
     sh(`doctoc ${file.path} --github --maxlevel 4`)
-      .then(() => sh(`git add ${file.path}`))
+      .then(() => insideGitRepo && sh(`git add ${file.path}`))
```
Comment (layershifter) [inline build/gulp/plugins/gulp-doctoc.ts:12]: We don't have Git repo in Now deployment

**Comment by layershifter** [comment]
We should also install [Now App](https://github.com/apps/now) to get it working in PR builds.

**Comment by kuzhelov** [comment]
@layershifter, could you, please, add information about usage plans that are provided by Zeit team for Now, as well as generally describe the workflow with this tool being introduced - directly to the PR description? Thank you!

**Review by bmdalex — APPROVED** [review]
👍 let's do this, try get another approval

**Comment by mnajdova** [comment]
@layershifter can we see as part of the PR how will the comment with the link to the served docs looks like? I think that is the only thing missing in the PR. Other then that looks good.

File: .gitignore [hunk 3]
```diff
@@ -1,4 +1,3 @@
-.awcache
```
Comment (layershifter) [inline .gitignore:1]: We don't use `awesome-typescript-loader` anymore...

**Pull request merged**